Source/Diagnostics/ComputeDiagFunctors/BackTransformFunctor.cpp

```diff
@@ -202,6 +228,64 @@ BackTransformFunctor::LorentzTransformZ (amrex::MultiFab& data, amrex::Real gamm
         amrex::Array4< amrex::Real > arr = data[mfi].array();
         amrex::Real clight = PhysConst::c;
         amrex::Real inv_clight = 1.0_rt/clight;
+#ifdef WARPX_DIM_RZ
+        const int n_rz = WarpX::n_rz_azimuthal_modes;
+        const int n_rcomps = WarpX::ncomps;
+        amrex::ParallelFor( tbx,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k)
+            {
+            // arr(x,y,z,comp) has (2*num_rz_modes-1)*10 components
+            // If there are 2 modes, then Er_0_real, Er_1_real, Er_2_real are the first three components
+            // followed by three components each for Et, Ez, Br, Bt, Bz, jr, jt, jz, and rho
+                for (int imode = 0; imode < n_rz; ++imode) {
+                    int offset = 1;
+                    // number of components for a given mode
+                    int ncomp = 2;
+                    if (imode == 0) {
+                        offset = 0;
+                        // Mode 0 has only real part, thus ncomp = 1
+                        ncomp = 1;
+                    }
+                    for (int icomp = 0; icomp < ncomp; ++icomp) {
+                        // Back-transform the transverse electric and magnetic fields.
+                        // Note that the z-components, Ez, Bz, are not changed by the transform.
+                        amrex::Real e_lab, b_lab, j_lab, rho_lab;
+
+                        // Offset for a given mode, imode, and component, icomp (0-real, 1-imag)
+                        const int mode_comp_offset = 2*imode + offset*(icomp-1);
```

Comment:
If simplify the above loop as suggested, then we can remove these lines.
```suggestion
                        // Offset for a given mode, imode, and component, icomp (0-real, 1-imag)
                        const int mode_comp_offset = 2*imode + offset*(icomp-1);
```

Reply:
I wanted to keep it this way since it accesses the other fields easily
Or rather it has the same "structure" as for cartesian
```
                e_lab = gamma_boost * ( arr(i, j, k, 0)
                                        + beta_boost * clight * arr(i, j, k, 4) );
                b_lab = gamma_boost * ( arr(i, j, k, 4)
                                        + beta_boost * inv_clight * arr(i, j, k, 0) );
```
and for RZ we have 
```
                        e_lab = gamma_boost * ( arr(i, j, k, n_rcomps*0 + mode_comp_offset)
                                                + beta_boost * clight * arr(i, j, k, n_rcomps*4+ mode_comp_offset) );
                        b_lab = gamma_boost * ( arr(i, j, k, n_rcomps*4 + mode_comp_offset)
                                                + beta_boost * inv_clight * arr(i, j, k, n_rcomps*0 + mode_comp_offset) );
```

Reply:
But I am open to changing this as well.

Reply:
okay this is done :) 
Thank you so much @RemiLehe 
Your suggestion is more clean and readable! Thanks!